### PR TITLE
failpoint: avoid returning the undeclared error

### PR DESCRIPTION
**Pull request by shafreeck**
Signed-off-by: Shafreeck Sea <shafreeck@gmail.com>

### What problem does this PR solve? 
Some unuseful messages are printed

```
failpoint is disabled on xxx
```

### What is changed and how it works?

* Return errors directly for `failpoints.Eval`
* Add more checks before printing the log message
* Print error with the failpath

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: failpoints.go [hunk 1]
```diff
@@ -186,11 +186,7 @@ func (fps *Failpoints) Eval(failpath string) (Value, error) {
 		return nil, ErrNotExist
 	}
 
-	val, err := fp.Eval()
-	if err != nil {
-		return nil, fmt.Errorf("%v on %s", err, failpath)
-	}
-	return val, nil
+	return fp.Eval()
```
Comment (kennytm) [inline failpoints.go:189]: maybe use `fmt.Errorf("%w on %s", err, failpath)` or use `errors.Annotatef`?

then in `EvalContext` check using `errors.Unwrap` (for built-in errors) / `errors.Cause` (for pingcap/errors).

Comment (shafreeck) [reply failpoints.go:189]: `errors.Unwrap` is not supported by the old go versions. `errors.Cause` works, however, the error handling here is too simple to introduce the `pingcap/errors`.

I really prefer the solution in this PR, but if you insist, I can change to your way.

Comment (kennytm) [reply failpoints.go:189]: The problem of removing the `fmt.Errorf` is that the `failpath` is lost when using `(*Failpoints).Eval` directly.

I think it is safe to upgrade the CI's Go version to 1.13+, as the latest version is 1.14, and all TiDB projects are targeting 1.13+.

(BTW in the actual implementation we should use `errors.Is`, not `errors.Unwrap`)

Comment (shafreeck) [reply failpoints.go:189]: `failpath`  is not lost, it is supplied by the user, the user should handle the error and failpath together if he/she would like to.

It is not just a problem of CI, some other users may still use older go versions, there is no more benefit to just give up these users.

Comment (shafreeck) [reply failpoints.go:189]: @kennytm Do you have any more suggestions? Or are you still worry about the `failpath`?

Considering that we have many methods of type `Failpoints`, like `Status`, `Enable`, `Disable`, `List`... They all return the original error directly. I think it was a design mistake to return an error with the `failpath`. Now it's time to fix it. I am blocked on this PR for a long time, looking forward to your reply.

Comment (lonng) [reply failpoints.go:189]: All filtered errors are defined in the package `failpoint`, I think we can define a customized error type to attach the extra information.

Comment (shafreeck) [reply failpoints.go:189]: @lonng OK, going to try that.

Comment (shafreeck) [reply failpoints.go:192]: @lonng Use the FpError to indicates the internal errors

**Pull request merged**